salt/modules/aptly.py

```diff
@@ -0,0 +1,517 @@
+# -*- coding: utf-8 -*-
+'''
+Aptly Debian repository manager.
+
+.. versionadded:: Oxygen
+'''
+# Import python libs
+from __future__ import absolute_import
+import json
+import logging
+import os
+import re
+
+# Import salt libs
+from salt.exceptions import SaltInvocationError
+import salt.utils.path
+
+
+_DEFAULT_CONFIG_PATH = '/etc/aptly.conf'
+_LOG = logging.getLogger(__name__)
+
+# Define the module's virtual name
+__virtualname__ = 'aptly'
+
+
+def __virtual__():
+    '''
+    Only works on systems with the aptly binary in the system path.
+    '''
+    if salt.utils.path.which('aptly'):
+        return __virtualname__
+    return (False, 'The aptly binaries required cannot be found or are not installed.')
+
+
+def _cast_if_numeric(value):
```

Comment:
We could likely use this as a general utility function. Perhaps we should move this into the `salt.utils` family?

Reply:
@cachedout I just checked and `stringutils` does have something similar (`to_num`), though it doesn't handle empty strings the way I would like. I'll add a commit to add the functionality I need to `stringutils`.